tensorflow_datasets/image/div2k.py

```diff
@@ -0,0 +1,150 @@
+    "valid_realistic_wild_x4": _DL_URL + "DIV2K_valid_LR_wild.zip",
+}
+
+_DATA_OPTIONS = ["bicubic_x2", "bicubic_x3", "bicubic_x4", "bicubic_x8",
+                 "unknown_x2", "unknown_x3", "unknown_x4",
+                 "realistic_mild_x4", "realistic_difficult_x4",
+                 "realistic_wild_x4"]
+
+class Div2kConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for Div2k."""
+
+  def __init__(self, data, **kwargs):
+    """Constructs a Div2kConfig."""
+    if data not in _DATA_OPTIONS:
+      raise ValueError("data must be one of %s" % _DATA_OPTIONS)
+
+    name = kwargs.get("name")
+    if name is None:
+      name = data
+    kwargs["name"] = name
+
+    description = kwargs.get("description")
+    if description is None:
+      description = "Uses %s data." % data
+    kwargs["description"] = description
+
+    super(Div2kConfig, self).__init__(**kwargs)
+    self.data = data
+
+  def download_urls(self):
+    """Returns train and validation download urls for this config."""
+    urls = {
+        "train_lr_url": _DL_URLS["train_"+self.data],
+        "valid_lr_url": _DL_URLS["valid_"+self.data],
+        "train_hr_url": _DL_URLS["train_hr"],
+        "valid_hr_url": _DL_URLS["valid_hr"],
+    }
+    return urls
+
+def _make_builder_configs():
+  configs = []
+  for data in _DATA_OPTIONS:
+    configs.append(Div2kConfig(
+        version=tfds.core.Version("1.0.0"),
+        data=data))
+  return configs
+
+class Div2k(tfds.core.GeneratorBasedBuilder):
+  """DIV2K dataset: DIVerse 2K resolution high quality images"""
+
+  BUILDER_CONFIGS = _make_builder_configs()
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "lr": tfds.features.Image(),
+            "hr": tfds.features.Image(),
+        }),
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+
+    extracted_paths = dl_manager.download_and_extract(
+        self.builder_config.download_urls())
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                "lr_path": extracted_paths["train_lr_url"],
+                "hr_path": extracted_paths["train_hr_url"],
+            }
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            gen_kwargs={
+                "lr_path": extracted_paths["valid_lr_url"],
+                "hr_path": extracted_paths["valid_hr_url"],
+            }
+        ),
+    ]
+
+  def _generate_examples(self, lr_path, hr_path):
+    """Yields examples."""
+    if not tf.io.gfile.listdir(hr_path)[0].endswith(".png"):
+      hr_path = os.path.join(hr_path, tf.io.gfile.listdir(hr_path)[0])
+
+    for root, dirs, files in tf.io.gfile.walk(lr_path):
+      if len(files) == 0:
+        continue
+      for file in files:
+        yield root + file, {
```

Comment:
Also why are you yielding `root + file`, shouldn't you be yielding only the dictionary?